Remove unused dependencies

nelstrom opened the pull request:
This addon was looking quite out of date. In particular, the fact that it depended on `ember-cli-babel` version 5 was starting to cause issues for an Ember app that depends on this addon.

These changes are fairly substantial, so I bumped the version from `0.2.0` to `1.0.0`. I don't actually believe that there were any breaking changes, so maybe `0.3.0` would have been appropriate?

Here's a summary of what I changed:

## Consolidating dependencies

The main functionality of this addon is implemented in the `index.js` file. That file has exactly two dependencies:

* `'git-repo-info'`
* `'ember-cli-deploy-plugin'`

It makes sense to declare these as `dependencies` in the `package.json` file. That leaves the following as `devDependencies`:

* `'chai-as-promised'`
* `'chai'`
* `'ember-cli'` (required from `ember-cli-build.js`)
* `'glob'`
* `'mocha-jshint'`
* `'mocha'`

These are mostly used for running unit tests.

Everything else must go! 🔥🔥🔥 

## Use native Promise instead of RSVP

Node has had Promise API for ages. This addon supports node 10 and later. We don't need RSVP as a polyfill anymore.

## Update Travis configuration

The travis configuration was out of date (using node version 4!). I used the current version of ember-cli (`3.26`) to generate a scratch addon and copied the `.travis.yml` over to this addon, then made tweaks to get it working again.

## Remove ember-try

ember-try is great for testing addons against different versions of Ember. To use ember try, you have to run tests via `ember test`, which in turn requires `ember-source`, `ember-cli-htmlbars` and `ember-cli-babel` (Remember? That was how we got here in the first place!)... I don't think any of that stuff belongs in this addon.

I removed the `ember-try` addon from `package.json` and also removed the `config/ember-try.js` file.

## Remove bower

bower looks like a relic from a bygone era. We don't need it.

nelstrom committed:
Remove dependency on babel

nelstrom committed:
Use native Promise instead of RSVP

nelstrom committed:
Remove unused dependencies

nelstrom committed:
Upgrade dependencies

nelstrom committed:
Update jshintrc to allow native Promise

nelstrom committed:
Update .travis.yml

This is the version that is generated
when running `ember addon NAME`.

nelstrom committed:
Remove the lint step

The unit tests already run the linter.

nelstrom committed:
Simplify travis config

We don't need to run `ember test`, which requires ember-cli-babel and
ember-source and loads of other stuff.

We can just use `yarn test` which is lightweight
and appropriate for this addon.

nelstrom committed:
Remove bower

nelstrom committed:
Remove config/release.js

This file is for configuring https://github.com/shipshapecode/ember-cli-release

The addon is deprecated.

The README recommends using https://github.com/release-it/release-it

nelstrom committed:
Remove config file for ember-try

This addon doesn't have any ember specific code in it,
so we don't really need to test this addon against different
versions of Ember.

nelstrom committed:
Add yarn.lock file

achambers commented:
Thanks for this @nelstrom . You're a ⭐ 

Would you mind dropping the commit that bumps the version then one that reverts that commit? No point them being there. Other than that, It's good to go.

nelstrom commented:
@achambers done.

nelstrom committed:
Re-install ember-cli-babel

It seems as though ember-cli makes it a hard requirement
that ember-cli-babel is available as a preprocessor for .js files.

https://github.com/ember-cli/ember-cli/blob/70ae16c5eb972076580b23e9c22579234749862e/lib/models/addon.js#L1308-L1334

achambers commented on package.json:
```diff
@@ -18,40 +18,20 @@
-    "ember-cli-inject-live-reload": "^1.4.0",
-    "ember-cli-jshint": "^1.0.0",
-    "ember-cli-qunit": "^2.1.0",
-    "ember-cli-release": "1.0.0-beta.2",
```
FYI - this addon (and the corresponding `config/release.js` file) is used for publishing this addon to npm and updating GH releases. However, https://github.com/release-it/release-it is the more modern way of doing this these days and so this repo could probably do with updating to use that. So, we can keep this change for now. Just an FYI.

achambers commented:
Thanks @nelstrom . Published as https://github.com/achambers/ember-cli-deploy-git-revision-data/releases/tag/v1.0.0
